reanahub/reana
# helm: add possibility to generate self-signed certificate

When opening or closing interactive sessions, `Ingress` objects are
either created or deleted to correctly route requests to the responsible
containers. Each time this happens, Traefik reloads its configuration,
and its default self-signed certificate gets regenerated.

This causes problems with Jupyter notebooks, in particular when trying
to create a new WebSocket. Self-signed certificates are not considered
valid by browsers, unless users explicitly trust them. This means that,
when the certificate is regenerated, browsers do not consider it valid,
given that it was never trusted by the user. This causes the creation of
a new WebSocket connection to fail.

This commit makes it possible to generate a self-signed certificate on
deploy, which will then be used by Traefik, overriding its default one.

Closes reanahub/reana-workflow-controller#457

How to test:
1. Deploy REANA
2. Launch some workflows
3. Open and close some sessions
4. Open a new Jupyter notebook

Expected result: the notebook works correctly and there aren't any connection errors
